### DISCUSSION TITLE
how to retreive the directus_translations?

### DISCUSSION THREAD

**Discussion by lumos94** [Intevel/nuxt-directus#221] [Q&A]
Hello all,

I am trying to retrieve some custom translations from Directus.  I am getting an error back saying that `You don't have permission to access this`. 

Permissions are checked and I indeed have permission to access it (it works ok on another project I have which uses their SDK). Other collections are retrieved as expected.

```
const fetchFormTranslationss = async () => {
  try {
    const items = await getItems<FormTranslations>({
      collection: "directus_translations",
    });
  } catch (e) {
    console.error("Error fetching translations:", e);
  }
}
```

**Comment by sandros94** [comment]
This might sound redundant, but did you also test directly the Rest API endpoint with tools like `Postman`, `Hoppscotch`, `HTTPie`?

**Reply by Intevel** [reply]
Would be interesting

**Reply by lumos94** [reply]
Using `Postman` I call the `https://my-directus.com/translations` endpoint and I retrieve the translations ok. The module calls the  `https://my-directus.com/items/translations` that is why I am getting the above mentioned error. Is there a way that I can call the translations endpoint directly?

**Reply by Intevel** [reply]
You can use the useDirectus composable. But can you open an issue for that bug?

**Reply by lumos94** [reply]
Is there any documentation on how to use it?

**Reply by lumos94** [reply]
any updates?

**Comment by kheiner** [comment]
@lumos94 It looks like you're trying to request a feature (translations) that isn't currently supported by the module nor the `useDirectus` composable but it is supported by the `@directus/sdk`

Using `https://directus.pizza` instead of `https://my-directus.com`

[Directus Documentation: Custom Translations](https://docs.directus.io/reference/system/translations.html#list-translations) hits the endpoint `https://directus.pizza/translations` to get your desired response.

To my knowledge, there's nothing you can do to get the translations you're requesting via `getItems` in `nuxt-directus`.  Confirm this by testing your `Postman` request with same endpoint that getItems will use:

`https://directus.pizza/items/directus_translations` returns:
```json
{
    "errors": [
        {
            "message": "You don't have permission to access this.",
            "extensions": {
                "code": "FORBIDDEN"
            }
        }
    ]
}
```

I see three possible paths forward (ordered easiest first):
1. Use the REST API endpoint directly without engaging with the nuxt-directus or @directus/sdk by hitting the endpoint `/translations` to get your object.
2. Integrate the `@directus/sdk` into your server folder inside of NUXT and use the `readTranslations` (use this option of you have the translations secured behind permissions).
3. Create a composable that can be integrated into the `nuxt-directus` project so that others don't encounter a similar issue/confusion that you encountered.

**Reply by casualmatt** [reply]
As for the other discussion, any suggestion is most valuable, but it would make more sense if you could add it to the #271 discussion. 

When this module was built, there was no SDK or documentation for specific API endpoints, so that's why we have many edge cases. Then we tried to adopt the SDK last September, which was a terrible experience, but as I said, all the details are in #271.
